src/transformers/models/cpmant/modeling_cpmant.py

```diff
@@ -0,0 +1,1090 @@
+        use_cache: Optional[bool] = False,
+        dropout_p: Optional[float] = 0.0,
+        mask_modules: Optional[List[Tuple[bool, bool]]] = None,
+    ):
+
+        super().__init__()
+        self.num_layers = num_layers
+        self.use_cache = use_cache
+        if mask_modules is not None:
+            if len(mask_modules) == num_layers:
+                raise ValueError("The total number of masks should equal to num_layers")
+            for mask_module in mask_modules:
+                if len(mask_module) != 2:
+                    raise ValueError("For encoder, each mask should be (mask_att, mask_ffn)")
+        else:
+            mask_modules = [(False, False)] * num_layers
+
+        self.layers = nn.ModuleList(
+            [
+                CPMAntTransformerBlock(
+                    dim_model=dim_model,
+                    dim_ff=dim_ff,
+                    num_heads=num_heads,
+                    dim_head=dim_head,
+                    eps=eps,
+                    dropout_p=dropout_p,
+                    mask_att=mask_modules[ith][0],
+                    mask_ffn=mask_modules[ith][1],
+                    use_cache=self.use_cache,
+                )
+                for ith in range(num_layers)
+            ]
+        )
+
+        self.output_layernorm = CPMAntLayerNorm(dim_norm=dim_model, eps=eps)
+
+    def forward(
+        self,
+        hidden_states: torch.Tensor,
+        attention_mask: torch.Tensor,
+        position_bias: torch.Tensor,
+        past_key_values: Optional[List[Tuple[torch.Tensor, torch.Tensor]]] = None,
+    ):
+        """
+        Args:
+            hidden_states (`torch.Tensor`): input to the layer of shape `(batch, seq_enc, dim_model)`
+            attention_mask (`torch.Tensor`):
+                Avoid invalid areas to participate in the calculation of shape `(batch, seq_enc, seq_enc)`
+            position_bias (`torch.Tensor`):
+                Provides position information to attention mechanism of shape `(num_heads, seq_enc, seq_enc)`
+            past_key_value (`Tuple(torch.FloatTensor)`, *optional*): cached past key and value projection states
+        """  # noqa: E501
+        if not self.use_cache:
+            for layer in self.layers:
+                hidden_states = layer(hidden_states, attention_mask, position_bias)
+            hidden_states = self.output_layernorm(hidden_states)
+            return hidden_states
+
+        current_key_values = []
+        for i, module in enumerate(self.layers):
+            hidden_states = module(
+                hidden_states,
+                attention_mask,
+                position_bias,
+                past_key_value=past_key_values[i] if past_key_values else None,
+            )
+            current_key_values.append(hidden_states[1])
+            hidden_states = hidden_states[0]
+        hidden_states = self.output_layernorm(hidden_states)
+        return hidden_states, current_key_values
+
+
+class CPMAntIntermediate(nn.Module):
```

Comment:
This seems to be copied from `BertIntermediate`: https://github.com/huggingface/transformers/blob/6c62cfb2eff095c181481d8ae86c7f836b65d2d7/src/transformers/models/bert/modeling_bert.py#L439 can you add a `# Copied from` statement?